containers/base/Dockerfile

```diff
@@ -15,98 +15,97 @@
+EXPOSE 8080
+
+# Path configuration
+ENV PATH $PATH:/tools/node/bin:/tools/google-cloud-sdk/bin
+ENV PYTHONPATH /env/python
+
 # Setup OS and core packages
 RUN echo "deb-src http://ftp.us.debian.org/debian testing main" >> /etc/apt/sources.list && \
     apt-get update -y && \
-    apt-get install --no-install-recommends -y -q \
-        curl wget unzip git vim build-essential ca-certificates pkg-config \
-        libatlas-base-dev liblapack-dev gfortran \
-        libpng-dev libfreetype6-dev libxft-dev \
-        rsync \
-        libxml2-dev \
-        python2.7 python-dev python-setuptools python-zmq openssh-client && \
+    apt-get install --no-install-recommends -y -q python unzip ca-certificates build-essential \
+    libatlas-base-dev liblapack-dev gfortran libpng-dev libfreetype6-dev libxft-dev libxml2-dev \
+    python2.7 python-dev python-setuptools openssh-client wget git pkg-config && \
     easy_install pip && \
-    mkdir -p /tools
+    mkdir -p /tools && \
 
 # Save GPL source packages
-RUN mkdir -p /srcs && \
+    mkdir -p /srcs && \
     cd /srcs && \
     apt-get source -d python-zmq wget git ca-certificates pkg-config libpng-dev && \
     wget https://mirrors.kernel.org/gnu/gcc/gcc-4.9.2/gcc-4.9.2.tar.bz2 && \
-    cd /
+    cd / && \
 
 # Setup Python packages. Rebuilding numpy/scipy is expensive so we move this early
 # to reduce the chance that prior steps can cause changes requiring a rebuild.
-RUN pip install -U numpy==1.11.0 && \
-    pip install -U pandas==0.18.0 && \
-    pip install -U scipy==0.17.0 && \
-    pip install -U scikit-learn==0.17.1 && \
-    pip install -U sympy==0.7.6.1 && \
-    pip install -U statsmodels==0.6.1
-
-RUN pip install -U tornado==4.3 \
-                   pyzmq==15.2.0 \
-                   jinja2==2.8 \
-                   jsonschema==2.5.1 \
-                   python-dateutil==2.5.0 \
-                   pytz==2015.4 \
-                   pandocfilters==1.3.0 \
-                   pygments==2.1.3 \
-                   argparse==1.2.1 \
-                   mock==1.2.0 \
-                   requests==2.9.1 \
-                   oauth2client==2.0.2 \
-                   httplib2==0.9.2 \
-                   futures==3.0.5 && \
-    pip install -U matplotlib==1.5.1 && \
-    pip install -U ggplot==0.6.8 && \
-    pip install -U seaborn==0.7.0 && \
-    pip install -U notebook==4.2.0 && \
-    pip install -U PyYAML==3.11 && \
-    pip install -U six==1.9.0 && \
-    pip install -U ipywidgets==5.2.2 && \
-    pip install -U future==0.15.2 && \
-    pip install -U psutil==4.3.0 && \
-    pip install -U google-api-python-client==1.5.1  && \
-    pip install -U plotly==1.12.5 && \
-    pip install -U nltk==3.2.1 && \
-    pip install -U bs4==0.0.1 && \
-    pip install -U crcmod==1.7 && \
-    pip install -U pillow==3.4.1 && \
-    easy_install pip && \
-    find /usr/local/lib/python2.7 -type d -name tests | xargs rm -rf
+    pip install -U --no-cache-dir numpy==1.11.0 && \
```

Comment:
> "move them back to the end" --- does it have to be a different RUN group?

The pydatalab stuff I was referring to isn't a RUN command anyway, it's an "ADD". That being said, your comments about the cloudml SDK are also a good point.

In order to save space, we do have to remove the build-time dependency in the same step where we add it (otherwise it remains in the base image layer), but that does not mean the cloudml SDK has to be in the same step, since it does not seem to rely on any of those removed build-time dependencies. (i.e. I think we should move the cloudml SDK install to a subsequent step, and probably the same for tensorflow too).